pallets/settlement/src/lib.rs

```diff
@@ -813,7 +820,31 @@ decl_module! {
         #[weight = <T as Trait>::WeightInfo::execute_scheduled_instruction(*legs_count)]
         fn execute_scheduled_instruction(origin, instruction_id: u64, legs_count: u32) {
             ensure_root(origin)?;
-            Self::execute_instruction(instruction_id)?;
+            Self::execute_instruction_retryable(instruction_id)?;
+        }
+
+        /// Reschedules a failed instruction.
+        ///
+        /// # Arguments
+        /// * `instruction_id` - Target instruction id to reschedule.
+        ///
+        /// # Permissions
+        /// * Portfolio
+        ///
+        /// # Errors
+        /// * `InstructionNotFailed` - Instruction not in a failed state or does not exist.
+        #[weight = <T as Trait>::WeightInfo::change_receipt_validity()]
+        pub fn reschedule_instruction(origin, instruction_id: u64) {
+            let did = Identity::<T>::ensure_perms(origin)?;
+
+            ensure!(<InstructionDetails<T>>::get(instruction_id).status == InstructionStatus::Failed, Error::<T>::InstructionNotFailed);
+            <InstructionDetails<T>>::mutate(instruction_id, |details| details.status = InstructionStatus::Pending);
```

Comment:
Consider using `try_mutate` instead.

Reply:
What is the benefit of `try_mutate` in this case?

Reply:
The code becomes DRYer and you decode one time fewer so its more efficient.